homeassistant/components/traccar/__init__.py

```diff
@@ -1 +1,109 @@
-"""The traccar component."""
+"""Support for Traccar."""
+import logging
+
+import voluptuous as vol
+from aiohttp import web
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import HTTP_UNPROCESSABLE_ENTITY, HTTP_OK, CONF_WEBHOOK_ID
+from homeassistant.helpers import config_entry_flow
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.components.device_tracker import DOMAIN as DEVICE_TRACKER
+from .const import (
+    ATTR_ACCURACY,
+    ATTR_ALTITUDE,
+    ATTR_BATTERY,
+    ATTR_BEARING,
+    ATTR_ID,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    ATTR_SPEED,
+    ATTR_TIMESTAMP,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+TRACKER_UPDATE = "{}_tracker_update".format(DOMAIN)
+
+
+DEFAULT_ACCURACY = 200
+DEFAULT_BATTERY = -1
+
+
+def _id(value: str) -> str:
+    """Coerce id by removing '-'."""
+    return value.replace("-", "")
+
+
+WEBHOOK_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ID): _id,
+        vol.Required(ATTR_LATITUDE): cv.latitude,
+        vol.Required(ATTR_LONGITUDE): cv.longitude,
+        vol.Optional(ATTR_ACCURACY, default=DEFAULT_ACCURACY): vol.Coerce(float),
+        vol.Optional(ATTR_ALTITUDE): vol.Coerce(float),
+        vol.Optional(ATTR_BATTERY, default=DEFAULT_BATTERY): vol.Coerce(float),
+        vol.Optional(ATTR_BEARING): vol.Coerce(float),
+        vol.Optional(ATTR_SPEED): vol.Coerce(float),
+        vol.Optional(ATTR_TIMESTAMP): vol.Coerce(int),
+    }
+)
+
+
+async def async_setup(hass, hass_config):
+    """Set up the Traccar component."""
+    hass.data[DOMAIN] = {"devices": set(), "unsub_device_tracker": {}}
+    return True
+
+
+async def handle_webhook(hass, webhook_id, request):
+    """Handle incoming webhook with Traccar request."""
+    try:
+        data = WEBHOOK_SCHEMA(dict(request.query))
+    except vol.MultipleInvalid as error:
+        return web.Response(text=error.error_message, status=HTTP_UNPROCESSABLE_ENTITY)
+
+    attrs = {
+        ATTR_ALTITUDE: data.get(ATTR_ALTITUDE),
+        ATTR_BEARING: data.get(ATTR_BEARING),
+        ATTR_SPEED: data.get(ATTR_SPEED),
+    }
+
+    device = data[ATTR_ID]
+
+    async_dispatcher_send(
+        hass,
+        TRACKER_UPDATE,
+        device,
+        (data[ATTR_LATITUDE], data[ATTR_LONGITUDE]),
```

Comment:
Don't send latitude and longitude as a tuple. Please send them as separate arguments.